tests/cloner_test.go

```diff
@@ -2537,6 +2526,197 @@ var _ = Describe("all clone tests", func() {
 			Expect(annValue).To(Equal("true"))
 		})
 	})
+
+	var _ = Describe("Clone from volumesnapshot source", func() {
```

Comment:
How about some functional test for validation webhook?  clone to smaller pvc, etc?

Also test source initially not existing?:

Reply:
Right, added